eZ/Publish/API/Repository/Tests/FieldType/SearchBaseIntegrationTest.php

```diff
@@ -0,0 +1,1146 @@
+ * @version //autogentag//
+ */
+
+namespace eZ\Publish\API\Repository\Tests\FieldType;
+
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Operator;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Field;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\LogicalNot;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause\Field as FieldSortClause;
+use eZ\Publish\API\Repository\Values\Content\Search\SearchResult;
+use eZ\Publish\API\Repository\Tests\SetupFactory\LegacySolr;
+use eZ\Publish\API\Repository\Tests\SetupFactory\LegacyElasticsearch;
+
+/**
+ * Integration test for searching and sorting with Field criterion and Field sort clause.
+ *
+ * This abstract test case should be used as a base for a specific field type search
+ * integration tests. It will first create two Content objects with two distinct field
+ * values, then execute a series of tests for Field criterion and Field sort clause,
+ * explicitly limited on these two values.
+ *
+ * Field criterion will be tested for each supported operator in all possible ways,
+ * combined with LogicalNot criterion, while Field sort clause will be tested in
+ * ascending and descending order.
+ *
+ * Same set of tests will be executed for each type of search (Content search, Location
+ * search), and in case of a criterion separately for a filtering and querying type of
+ * Query.
+ *
+ * To get the test working extend it in a concrete field type test and implement
+ * methods:
+ *
+ * - getSearchValueA()
+ * - getSearchValueB()
+ *
+ * In the test descriptions Content object created with values A and B are referred to as
+ * Content A, and Content B. See the descriptions of the abstract declarations of these
+ * methods for more details on how to choose proper values.
+ *
+ * Note: this test case does not concern itself with testing field filters, behaviour
+ * of multiple sort clauses or combination with other criteria. These are tested
+ * elsewhere as a general field search cases, which enables keeping this test case
+ * simple.
+ */
+abstract class SearchBaseIntegrationTest extends BaseIntegrationTest
+{
+    /**
+     * Get field value A.
+     *
+     * The value must be valid for Content creation.
+     *
+     * When Field sort clause with ascending order is used on the tested field,
+     * Content containing the field with this value must come before the Content
+     * with value B.
+     *
+     * Opposite should be the case when using descending order.
+     *
+     * @return mixed
+     */
+    abstract protected function getSearchValueA();
```

Comment:
Ok, this turns out to be harder than expected, as I already use data providers for different operator cases with two test values, and PHPUnit can't combine data providers. This would require breaking existing fixtures per test that uses them, so ~150 new tests.